crates/pipeline-manager/src/compiler/sql_compiler.rs

```diff
@@ -0,0 +1,533 @@
+use crate::common_error::CommonError;
+use crate::compiler::util::{
+    cleanup_specific_directories, create_new_file, create_new_file_with_content, read_file_content,
+    recreate_dir, CleanupDecision,
+};
+use crate::config::{CommonConfig, CompilerConfig};
+use crate::db::error::DBError;
+use crate::db::storage::Storage;
+use crate::db::storage_postgres::StoragePostgres;
+use crate::db::types::common::Version;
+use crate::db::types::pipeline::PipelineId;
+use crate::db::types::program::{
+    generate_program_info, ProgramConfig, ProgramInfo, SqlCompilerMessage,
+};
+use crate::db::types::tenant::TenantId;
+use feldera_types::program_schema::ProgramSchema;
+use log::{debug, error, info, trace};
+use std::path::Path;
+use std::time::Instant;
+use std::{process::Stdio, sync::Arc};
+use tokio::{
+    fs,
+    process::Command,
+    sync::Mutex,
+    time::{sleep, Duration},
+};
+use uuid::Uuid;
+
+/// The frequency at which the compiler polls the database for new SQL compilation requests.
+/// It balances resource consumption due to polling and a fast SQL compilation response.
+const POLL_INTERVAL: Duration = Duration::from_millis(250);
```

Comment:
seems excessive? 1sec?

Reply:
The reason for the frequent polling is that:
(1) It directly adds to the user getting feedback on SQL compilation, hence 1 second would mean 1 second of added latency
(2) There is only 1 SQL compiler and 1 Rust compiler currently, as such the strain on the database seems limited with 4+4 requests/s.

Does the above seem alright? Or would it be too much strain / is there another consideration?

Reply:
I'll set the polling of Rust compiler every half a second.

Reply:
In future work, there would be a database notification mechanism.

Reply:
@snkas is there a reason we're polling instead of using the listen/notify mechanism?

Reply:
Ideally, we'd use the listen/notify mechanism instead of polling.

Reply:
Note that the UI also polls every second (I think?) for compilation status, which can make this feel even less responsive if the polling interleavings get unlucky.

Reply:
Yes, it's an unfortunate side-effect of async compilation. With the listen/notify mechanism, only UI polling should be effectively left.

In the "Field filters for API server endpoints GET pipeline(s) " it will also provide a light-weight endpoint that can be more frequently polled by the UI.